modules/core/src/test/java/org/apache/ignite/internal/GridMBeansTest.java

```diff
@@ -76,11 +83,83 @@ public void testExecutorBeans() throws Exception {
+        checkBeanName("Thread Pools", "GridQueryExecutor");
+        checkBeanName("Thread Pools", "GridSchemaExecutor");
+        checkBeanName("Thread Pools", "StripedExecutor");
+
+        // kernal beans
+        checkBeanName("Kernal", "IgniteKernal");
+        checkBeanName("Kernal", "ClusterLocalNodeMetricsMXBeanImpl");
+        checkBeanName("Kernal", "ClusterMetricsMXBeanImpl");
+    }
+
+    /** Check that kernal beans were successfully registered by their expected names */
+    public void testIgniteKernalBeansRegistration() throws Exception {
+        // standard executors
+        checkBeanRegistrationByName("Thread Pools", "GridExecutionExecutor");
+        checkBeanRegistrationByName("Thread Pools", "GridSystemExecutor");
+        checkBeanRegistrationByName("Thread Pools", "GridManagementExecutor");
+        checkBeanRegistrationByName("Thread Pools", "GridClassLoadingExecutor");
+        checkBeanRegistrationByName("Thread Pools", "GridQueryExecutor");
+        checkBeanRegistrationByName("Thread Pools", "GridSchemaExecutor");
+        checkBeanRegistrationByName("Thread Pools", "StripedExecutor");
+
+        // kernal beans
+        checkBeanRegistrationByName("Kernal", "IgniteKernal");
+        checkBeanRegistrationByName("Kernal", "ClusterLocalNodeMetricsMXBeanImpl");
+        checkBeanRegistrationByName("Kernal", "ClusterMetricsMXBeanImpl");
+    }
+
     /** Checks that a bean with the specified group and name is available and has the expected attribute */
     private void checkBean(String grp, String name, String attributeName, Object expAttributeVal) throws Exception {
         ObjectName mBeanName = IgniteUtils.makeMBeanName(grid().name(), grp, name);
         Object attributeVal = grid().configuration().getMBeanServer().getAttribute(mBeanName, attributeName);
 
         assertEquals(expAttributeVal, attributeVal);
     }
+
+    /** Checks that a bean name generation returns the expected value */
+    private void checkBeanName(String grp, String name) throws Exception {
+        final String EXPECTED_BEAN_NAME = getExpectedBeanName(JMX_DOMAIN, grid().name(), grp, name);
+
+        ObjectName mBeanName = IgniteUtils.makeMBeanName(grid().name(), grp, name);
+
+        assertTrue(EXPECTED_BEAN_NAME.equals(mBeanName.getCanonicalName()));
+    }
+
+    /** Checks that a bean name generation returns the expected value */
+    private void checkBeanRegistrationByName(String grp, String name) throws Exception {
+        final String EXPECTED_BEAN_NAME = getExpectedBeanName(JMX_DOMAIN, grid().name(), grp, name);
+
+        assertTrue(grid().configuration().getMBeanServer().isRegistered(new ObjectName(EXPECTED_BEAN_NAME)));
+    }
+
+    private String getExpectedBeanName(String domain, String instanceName, String grp, String name) {
+        StringBuffer sb = new StringBuffer();
+
+        sb.append(domain + ":");
+
+        if (getBoolean(IGNITE_MBEAN_APPEND_CLASS_LOADER_ID, true)) {
```

Comment:
Would it be possible to set both IGNITE_MBEAN_APPEND_CLASS_LOADER_ID and IGNITE_MBEAN_APPEND_JVM_ID to false, and then just hardcode the entire name string like "igniteInstanceName=mygrid,group=Thread Pools,name=GridSystemExecutor"? Then I guess it would be possible to check these names in the checkBean() test using them instead of makeMBeanName() call (basically, I'd rather get rid of the makeMBeanName() in this test).

Reply:
I will check.

Reply:
done